staging/src/kubevirt.io/client-go/kubecli/handler.go

```diff
@@ -210,65 +214,53 @@ func (v *virtHandlerConn) Pod() (pod *v1.Pod, err error) {
 	return v.pod, err
 }
 
-func (v *virtHandlerConn) Put(url string, tlsConfig *tls.Config, body io.ReadCloser) error {
+func (v *virtHandlerConn) doRequest(req *http.Request) (response string, err error) {
+	resp, err := v.httpClient.Do(req)
+	if err != nil {
+		return
+	}
 
-	client := http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: tlsConfig,
-		},
-		Timeout: 10 * time.Second,
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return "", fmt.Errorf("unexpected return code %s", resp.Status)
 	}
 
-	req, err := http.NewRequest(http.MethodPut, url, body)
+	defer resp.Body.Close()
```

Comment:
I think this should go right after `httpClient.Do` and error checking:

```go
	resp, err := v.httpClient.Do(req)
	if err != nil {
		return
	}
        defer resp.Body.Close()
```
Even if we get some weird `resp.StatusCode`, the connection still needs to be closed (now it is leaked).

Reply:
:+1:
Done